spec.html

```diff
@@ -34622,38 +34637,142 @@ <h1>%TypedArray%.prototype.filter ( _callbackfn_ [ , _thisArg_ ] )</h1>
+        <p>When the `indexOf` method is called with one or two arguments, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Perform ? ValidateTypedArray(_O_).
+          1. Let _len_ be _O_.[[ArrayLength]].
+          1. If _len_ is 0, return *-1*<sub>𝔽</sub>.
+          1. Let _n_ be ? ToIntegerOrInfinity(_fromIndex_).
+          1. Assert: If _fromIndex_ is *undefined*, then _n_ is 0.
+          1. If _n_ is +&infin;, return *-1*<sub>𝔽</sub>.
+          1. Else if _n_ is -&infin;, set _n_ to 0.
+          1. If _n_ &ge; 0, then
+            1. Let _k_ be _n_.
+          1. Else,
+            1. Let _k_ be _len_ + _n_.
+            1. If _k_ &lt; 0, set _k_ to 0.
+          1. Repeat, while _k_ &lt; _len_,
+            1. Let _elementK_ be ? Get(_O_, ! ToString(𝔽(_k_))).
+            1. Let _same_ be the result of performing Strict Equality Comparison _searchElement_ === _elementK_.
+            1. If _same_ is *true*, return 𝔽(_k_).
+            1. Set _k_ to _k_ + 1.
+          1. Return *-1*<sub>𝔽</sub>.
+        </emu-alg>
+        <p>This function is not generic. The *this* value must be an object with a [[TypedArrayName]] internal slot.</p>
       </emu-clause>
 
       <emu-clause id="sec-%typedarray%.prototype.join">
         <h1>%TypedArray%.prototype.join ( _separator_ )</h1>
-        <p>%TypedArray%`.prototype.join` is a distinct function that implements the same algorithm as `Array.prototype.join` as defined in <emu-xref href="#sec-array.prototype.join"></emu-xref> except that the *this* value's [[ArrayLength]] internal slot is accessed in place of performing a [[Get]] of *"length"*. The implementation of the algorithm may be optimized with the knowledge that the *this* value is an object that has a fixed length and whose <emu-xref href="#integer-index">integer-indexed</emu-xref> properties are not sparse. However, such optimization must not introduce any observable changes in the specified behaviour of the algorithm.</p>
-        <p>This function is not generic. ValidateTypedArray is applied to the *this* value prior to evaluating the algorithm. If its result is an abrupt completion that exception is thrown instead of evaluating the algorithm.</p>
+        <p>The interpretation and use of the arguments of %TypedArray%`.prototype.join` are the same as for `Array.prototype.join` as defined in <emu-xref href="#sec-array.prototype.join"></emu-xref>.</p>
+        <p>When the `join` method is called with one argument _separator_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Perform ? ValidateTypedArray(_O_).
+          1. Let _len_ be _O_.[[ArrayLength]].
+          1. If _separator_ is *undefined*, let _sep_ be the single-element String *","*.
+          1. Else, let _sep_ be ? ToString(_separator_).
```

Comment:
In a similar vein, with this PR, `ta.join(detachPoison)` is now specced to return a string with a length-1 number of `,`s. V8 already matches this, SM prints a bunch of `undefined`.

```javascript
function detach(ab) {
  if (ArrayBuffer.transfer) {
    ArrayBuffer.transfer(ab);
  } else if (ArrayBuffer.detach) {
    ArrayBuffer.detach(ab);
  } else if (typeof detachArrayBuffer === "function") {
    detachArrayBuffer(ab);
  } else if (typeof transferArrayBuffer === "function") {
    transferArrayBuffer(ab)
  } else if (typeof Worker === "function") {
    try { eval("%ArrayBufferDetach(ab)") } catch (e) {
      var w = new Worker("", {type: "string"});
      w.postMessage(ab, [ab]);
      w.terminate();
    }
  } else {
    throw new TypeError("cannot detach array buffer");
  }
}

let ta = new Uint8Array(2);
print(ta.join({ toString() { detach(ta.buffer); return ','; } }));
```

```
#### jsc

TypeError: Underlying ArrayBuffer has been detached from the view

#### sm
undefined,undefined

#### v8
,
```
```